Kudu.Core/Infrastructure/VsSolutionProject.cs

```diff
@@ -161,22 +161,11 @@ private void EnsureProperties()
             {
                 // used to determine project type
                 _projectTypeGuids = VsHelper.GetProjectTypeGuids(_absolutePath);
-                if (AspNetCoreHelper.IsDotnetCoreFromProjectFile(_absolutePath, _projectTypeGuids))
-                {
-                    _isAspNetCore = true;
-                }
-                else if (projectType == SolutionProjectType.KnownToBeMSBuildFormat)
-                {
-                    // KnownToBeMSBuildFormat: C#, VB, and VJ# projects
-                    // Check if it's a wap
-                    _isWap = VsHelper.IsWap(_projectTypeGuids);
-
-                    _isExecutable = VsHelper.IsExecutableProject(_absolutePath);
-                }
-                else if (FunctionAppHelper.LooksLikeFunctionApp())
-                {
-                    _isFunctionApp = true;
-                }
+
+                _isAspNetCore = AspNetCoreHelper.IsDotnetCoreFromProjectFile(_absolutePath, _projectTypeGuids);
+                _isWap = VsHelper.IsWap(_projectTypeGuids);
+                _isExecutable = VsHelper.IsExecutableProject(_absolutePath);
+                _isFunctionApp = FunctionAppHelper.LooksLikeFunctionApp(Path.GetDirectoryName(_absolutePath));
```

Comment:
1.`isAspNetCore` check if the csproj has no Guid
2.`isWap` check if the  csproj contains a particular Guid
3.`isExecutable` check if the csproj has outputtype == exe
4.`isFunctionApp` check if the project has host.json and environment variable `FUNCTION_EXTENSION_VERSION`

1,4 or 1,3 can be true at the same time, but neither of them have our support in Azure